pkg/webhooks/ingresscontroller/ingresscontroller.go

```diff
@@ -149,12 +176,151 @@ func (wh *IngressControllerWebhook) authorized(request admissionctl.Request) adm
+		if err != nil {
+			log.Error(err, "err parsing machineCIDR into network cidr", "machineCIDR", string(instConf.Networking.MachineCIDR))
+			return nil, nil, err
+		}
+		if machIP == nil || machNet == nil {
+			err := fmt.Errorf("failed to parse machineCIDR string:'%s' into network structures", string(instConf.Networking.MachineCIDR))
+			log.Error(err, "failed to parse install-config machineCIDR")
+			return nil, nil, err
+		}
+		// Successfully fetched, parsed, and converted the machineCIDR string into net structures...
+		wh.machineCIDRIP = machIP
+		wh.machineCIDRNet = machNet
+	}
+	return wh.machineCIDRIP, wh.machineCIDRNet, nil
+}
+
+/* Fetch the install-config from the kube-system config map's data.
+ * this requires proper role, rolebinding for this service account's get() request
+ * to succeed. (see toplevel selectorsyncset).  This config should not change during runtime so
+ * this operation should cache this if possible.
+ * TODO: Should it retry fetching the config if there are any failures/errors encountered while
+ * parsing out the the desired values?
+ */
+func (wh *IngressControllerWebhook) getClusterConfig() (*installConfig, error) {
+	var err error
+	if wh.kubeClient == nil {
+		wh.kubeClient, err = k8sutil.KubeClient(&wh.s)
+		if err != nil {
+			log.Error(err, "Fail creating KubeClient for IngressControllerWebhook")
+			return nil, err
+		}
+	}
+	clusterConfig := &corev1.ConfigMap{}
+	err = wh.kubeClient.Get(context.Background(), client.ObjectKey{Name: installConfigMap, Namespace: installConfigNamespace}, clusterConfig)
+	if err != nil {
+		log.Error(err, "Failed to fetch configmap: 'cluster-config-v1' for cluster config")
+		return nil, err
+	}
+	data, ok := clusterConfig.Data[installConfigKeyName]
+	if !ok {
+		return nil, fmt.Errorf("did not find key %s in configmap %s/%s", installConfigKeyName, installConfigNamespace, installConfigMap)
+	}
+	instConf := &installConfig{}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader([]byte(data)), 4096)
+	if err := decoder.Decode(instConf); err != nil {
+		return nil, errors.Wrap(err, "failed to decode install config")
+	}
+	return instConf, nil
+}
+
+func (wh *IngressControllerWebhook) checkAllowsMachineCIDR(ipRanges []operatorv1.CIDR) admissionctl.Response {
+	// https://docs.openshift.com/container-platform/4.13/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-load-balancer-allowed-source-ranges.html
+	// Note: From docs it appears a missing ASR value/attr allows all. However...
+	// once ASR values have been added to an ingresscontroller, later deleting all the ASRs can expose an issue
+	// where the IGC will remaining in progressing state indefinitely.
+	// For now return Allowed, but with a warning?
+	if ipRanges == nil || len(ipRanges) <= 0 {
+		return admissionctl.Allowed("Allowing empty 'AllowedSourceRanges'. Populate this value if operator remains in 'progressing' state")
```

Comment:
```suggestion
		return admissionctl.Allowed("Allowing empty 'AllowedSourceRanges'")
```

IMO, we should avoid suggesting configuration changes in component logs; the product documentation should be the source-of-truth we provide to customers